src/rebar3_hex.erl

```diff
@@ -60,16 +61,21 @@ task_args(State) ->
             {Task, proplists:delete(task, Opts)}
     end.
 
+
+%% TODO: A current limitation of rebar3_hex:task_state/1 is it returns strings.
```

Comment:
In truth, we'll probably always need to return these. `rebar3 hex retire` is a good example of where this pattern simply breaks down.